ext/standard/tests/mail/mail_variation_win.phpt

```diff
@@ -0,0 +1,86 @@
+--TEST--
+Test mail() function : basic functionality (win)
+--EXTENSIONS--
+curl
+--SKIPIF--
+<?php require_once __DIR__.'/mail_windows_skipif.inc'; ?>
+--INI--
+SMTP=localhost
+smtp_port=1025
+--FILE--
+<?php
+
+require_once __DIR__.'/mailhog_utils.inc';
+
+$from = ini_get('sendmail_from');
+
+$cases = [
+    [
+        'label' => 'From is not set',
+        'premise' => function () {
+            ini_set('SMTP', 'localhost');
+            ini_set('smtp_port', 1025);
+        }
+    ],
+    [
+        'label' => 'Invalid port',
+        'premise' => function () {
+            ini_set('SMTP', 'localhost');
+            ini_set('smtp_port', 2525);
+            ini_set('sendmail_from', 'from@example.com');
+        }
+    ],
+    [
+        'label' => 'Invalid host',
+        'premise' => function () {
+            ini_set('SMTP', 'localplace');
+            ini_set('smtp_port', 1025);
+            ini_set('sendmail_from', 'from@example.com');
+        }
+    ],
+];
+
+foreach ($cases as $index => ['label' => $label, 'premise' => $premise]) {
+    echo "========== {$label} ==========\n";
+
+    $premise();
+
+    $to = "mail_variation_win_{$index}@example.com";
+    $subject = "{$index}: Basic PHPT test for mail() function";
+    $message = <<<HERE
+Description
+bool mail ( string \$to , string \$subject , string \$message [, string \$additional_headers [, string \$additional_parameters]] )
+Send an email message
+HERE;
+
+    $res = mail($to, $subject, $message);
+
+    if ($res === true) {
+        echo "Sent the email.\n";
+    }
+
+    $res = searchEmailByToAddress($to);
+
+    if (
+        $res &&
+        getFromAddress($res) === $from &&
+        getToAddresses($res)[0] === $to &&
+        getSubject($res) === $subject &&
+        getBody($res) === $message
```

Comment:
as this is pretty much repeated everywhere, it might make sense to have a a function in utils for this. Maybe something like:
```php
mailCheckResponse($res, $from, $to, $subject, $message)
```

Reply:
Makes sense. I'll make it a function.